### ISSUE TITLE
Request review of Keyboard Lock

### ISSUE THREAD

**Issue by garykac** [w3ctag/design-reviews#192]
Hello TAG!

I'm requesting a TAG review of:

  - Name: Keyboard Lock
  - Specification URL: [https://w3c.github.io/keyboard-lock/](https://w3c.github.io/keyboard-lock/)
  - Tests: Requires manual tests for keyboard input
  - Primary contacts: garykac, jamiewalch

Further details (optional):

  - Relevant time constraints or deadlines: Before end of Sept
  - [X] I have read and filled out the [Self-Review Questionnare on Security and Privacy](https://www.w3.org/TR/security-privacy-questionnaire/). The [assessment is here](https://github.com/w3c/keyboard-lock/blob/gh-pages/tag-security-privacy.md).
  - [X] I have reviewed the TAG's [API Design Principles](https://w3ctag.github.io/design-principles/)

You should also know that...

We'd prefer the TAG provide feedback as (please select one):

  - [X] open issues in our Github repo for each point of feedback

**Comment by annevk** [comment]
Filed https://github.com/w3c/keyboard-lock/issues/32 and https://github.com/w3c/keyboard-lock/issues/33, though not on behalf of the TAG.

**Comment by cynthia** [comment]
Couple more in w3c/keyboard-lock#34 and w3c/keyboard-lock#35, although those are mostly open ended.

**Comment by dbaron** [comment]
So we started comparing lock APIs in today's meeting:
* [Wake Lock](https://w3c.github.io/wake-lock/) has `navigator.getWakeLock("screen").then(wakeLock => wakeLock.createRequest())`
* [Screen Orientation API](https://w3c.github.io/screen-orientation/) has `screen.orientation.lock('portrait')`
* [Pointer Lock](https://w3c.github.io/pointerlock/) has `element.requestPointerLock()` and then fires events
* [Keyboard Lock](https://w3c.github.io/keyboard-lock/) now proposes `navigator.requestKeyboardLock().then(...)`

**Comment by cynthia** [comment]
So we did a breakout during our F2F, and the design looks fine. (Aside from the issues raised above)

However we do have some remarks on the naming of the spec - we believe that 1) either keyboard can be a global object (e.g. navigator.keyboard) and there is a `lock()/unlock()` within that for extensibility, or 2) for logical naming consistency potentially `keyboardLock()/keyboardUnlock()` and remove the request part. This is mostly for the purpose of being consistent with other stable platform features like the screen orientation lock mentioned above.

(It would also be nice if the pointer lock spec could have a newer promise based method too following this convention, but that's out of scope for this review)

**Comment by garykac** [comment]
I wanted to follow-up on some changes that we made in reaction to @cynthia 's comments about naming.

Initially, we opted for naming option (2) `keyboardLock`/`keyboardUnlock` since we thought it would be overkill to have a `navigator.keyboard` object that was only used for locking and unlocking.

However, we are now aware of at least one other API proposal that makes sense to put on a `keyboard` object, so we're switching to (1) creating `navigator.keyboard` and having `lock` and `unlock` methods. This is consistent with the naming of the Screen Orientation API.

I don't believe this is contentious, but I wanted to bring it up in case there are any more comments on the naming. Thanks!
